app/controllers/application_controller.rb

```diff
@@ -24,4 +24,8 @@ def merchant_admin?
   def site_admin?
     current_user&.site_admin?
   end
+
+  def render_404
+    render file: '/public/404'
+  end
```

Comment:
Good use of this helper method. Makes our code look clean, and we can change this in the future if needed without editing a lot code.